src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```diff
@@ -0,0 +1,210 @@
+package edu.berkeley.ce.rockslicing
+
+import breeze.linalg
+import breeze.linalg.DenseVector
+import scala.annotation.tailrec
+
+object JointGenerator {
+  /**
+    * Finds the normal to the joint specified by the input strike and dip
+    *
+    * @param strike Strike of the joint plane given in degrees
+    * @param dip Dip of the joint plane given in degrees
+    * @return Array that specifies the x, y and z components of the joint plane
+    */
+  private def findJointNormal(strike: Double, dip: Double): Array[Double] = {
+    val strikeRadians = strike.toRadians
+    val dipRadians = dip.toRadians
+    val strikeVector = DenseVector[Double](math.cos(-strikeRadians), math.sin(-strikeRadians), 0.0)
+    // If joint is vertical, dip vector will point along negative z-axis
+    val dipVector = if (math.abs(dip - 90.0) < NumericUtils.EPSILON) {
+      DenseVector[Double](0.0, 0.0, -1.0)
+    } else {
+      DenseVector[Double](math.cos(dipRadians) * math.cos(-(strikeRadians + math.Pi / 2.0)),
+        math.cos(dipRadians) * math.sin(-(strikeRadians + math.Pi / 2.0)),
+        -math.sin(dipRadians))
+    }
+    val jointNormal = linalg.normalize(linalg.cross(strikeVector, dipVector))
+    Array(jointNormal(0), jointNormal(1), jointNormal(2))
+  }
+
+  /**
+    * Finds the faces that define the rock volume of interest
+    *
+    * @param globalOrigin Coordinates of global origin
+    * @param rockVolumeData Array of arrays describing each of the faces that define the boundaries of
+    *                       the rock volume of interest. The inputs in each array are strike, dip and the x,
+    *                       y and z coordinates of a point that lies within the plane containing the face.
+    *                       Strike and dip should be specified in degrees.
+    * @return A sequence of faces that represent the rock volume of interest
+    */
+  private def findBoundingFaces(globalOrigin: Array[Double], rockVolumeData: Seq[Array[Double]]): Seq[Face] = {
+    rockVolumeData map { parameters =>
+      val faceCenter = Array(parameters(2), parameters(3), parameters(4))
```

Comment:
Yeah, seeing this code makes me think we should express the rock volume as a collection of `Face` instances. This code would become *much* more readable. For example:
```scala
rockVolume map { case Face(strike, dip, center) =>
    // You don't even need an explicit faceCenter any more
    val normal = findJointNormal(strike, dip)
```
But this chunk of code also reminded me that our `Face` objects are slightly different in that they are defined by normal vector and distance rather than center. So maybe we need a separate (and very simple) case class for the input faces?

I just think we should use the benefits of higher level programming languages, i.e. giving human readable names to things, where they make for clearer code and don't cost us much.